src/Mvc/Mvc.RazorPages/src/ApplicationModels/PageApplicationModel.cs

```diff
@@ -154,5 +156,10 @@ public PageApplicationModel(PageApplicationModel other)
         /// Gets the sequence of <see cref="PagePropertyModel"/> instances on <see cref="PageHandlerModel"/>.
         /// </summary>
         public IList<PagePropertyModel> HandlerProperties { get; }
+
+        /// <summary>
+        /// Gets the endpoint metadata for this action.
+        /// </summary>
+        public IList<object> EndpointMetadata { get; }
```

Comment:
Adding this because there wasn't a way to do this before? I thought we had this on pageroutemodel?

Reply:
Yes, we have `EndpointMetadata` on `PageRouteModel`. However, here's how things work with Razor Pages:

`PageRouteModel` -> `PageActionDescriptor` -> `PageApplicationModel` -> `Apply conventions` -> `CompiledPageActionDescriptor`

When going from `PageRouteModel` to `PageActionDescriptor`: we've already turned individual selector models in to individual `PageActionDescriptor` instances. Secondly, we cannot expose selector model on `PageApplicationModel` since it's too late to influence the route, but you do get to influence route metadata.

Reply:
okie dokie.